src/utils/chart.ts

```diff
@@ -0,0 +1,97 @@
+import { AppAction } from '@graasp/sdk';
+
+import {
+  addMinutes,
+  addSeconds,
+  format,
+  formatDuration,
+  getMinutes,
+  intervalToDuration,
+  isBefore,
+  isEqual,
+  parseISO,
+  setMinutes,
+} from 'date-fns';
+
+import { CodeVersionType } from '@/interfaces/codeVersions';
+
+const INTERVAL_DURATION_IN_MINUTES = 10;
+export const VERSION_STEP_DURATION = 2000;
+
+// Function that take starttime and end time and handle intervals between depends on INTERVAL_DURATION_IN_MINUTES
+export const distributeIntervals = ({
+  startDate,
+  endDate,
+}: {
+  startDate: string;
+  endDate: string;
+}): { [key: string]: number } => {
+  const start = parseISO(startDate);
```

Comment:
If I understand correctly, the start time is defined by the fist action and will not be the same for all users right ? Would it be easier to use fixed intervals starting for example on the hour ? like 01:00, 01:10, 01:20 ? I don't think it will make a big difference in the visualisation, but it should allow for easier computations since they will be mathematically defined, so you not need to materialize them ?